### PR TITLE
[meta] feature: protobuf message has to persist `MIN_COMPATIBLE_VER` in it

### PR DESCRIPTION
**Pull request by drmingdrmer**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

##### [meta] feature: protobuf message has to persist `MIN_COMPATIBLE_VER` in it

**Non-breaking** meta-protobuf change.

This way to let old query be able to decide if it is safe to load data
written by a newer query executable.

- Fix: #5784

## Changelog

- New Feature

## Related Issues

### CODE REVIEW

**Comment by BohuTANG** [comment]
This is a breaking change?

**Comment by drmingdrmer** [comment]
> This is a breaking change?

No:)

This PR only affects the query:
It just added a new field into protobuf message. Old query will just ignore this new field when loading new data. New query will have a default value `0` when loading the old data.

File: common/protos/proto/config.proto [hunk 1]
```diff
@@ -18,6 +18,7 @@ package databend_proto;
 
 message S3StorageConfig {
   uint64 version = 100;
+  uint64 min_compatible = 101;
```
Comment (zhyass) [inline common/protos/proto/config.proto:21]: Why `min_compatible` is 101 bigger then `version`?

Comment (drmingdrmer) [reply common/protos/proto/config.proto:21]: This is the index of a field. not the value of it

Comment (zhyass) [reply common/protos/proto/config.proto:21]: 😂, My fault.

**Review by zhyass — APPROVED** [review]
LGTM

**Pull request merged**